internal/search/parser_test.go

```diff
@@ -1,12 +1,82 @@
 package search
 
 import (
+	"encoding/json"
 	"strings"
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
 )
 
+func Test_ScanParameter(t *testing.T) {
+	cases := []struct {
+		Name  string
+		Input string
+		Want  string
+	}{
+		{
+			Name:  "Normal field:value",
+			Input: `field:value`,
+			Want:  `{"field":"field","value":"value","negated":false}`,
+		},
```

Comment:
Tiniest nitpick: I would use real fields and real values in the test input, so it's easier to read and understand what's parsed into what.

```suggestion
		{
			Name:  "Normal field:value",
			Input: `repohasfile:README.md`,
			Want:  `{"field":"repohasfield","value":"README.md","negated":false}`,
		},
```

Reply:
![](https://media3.giphy.com/media/l49JRQC9RNa5j35a8/giphy.gif?cid=5a38a5a2a089452425a40defc62b24c5bffb7a39ef8c5934&rid=giphy.gif)

rewrite_all_the_test_inputs.gif

Reply:
>  I would use real fields and real values in the test input

I think part of the issue is that there aren't good real values for the things I'm testing:

`field:value:value`, `-:value`, ... but I'll try.